src/components/BoardConfig.jsx

```diff
@@ -20,6 +20,14 @@ const BoardConfigForm = styled.div`
   }
 `;
 
+const filterEmptyArraysAndFalsyNonBooleanValues = (dataObject) => {
```

Comment:
Did you try to avoid empty values instead:

```js
            <Field
              name="playerCount"
              initialValue={boardConfig.playerCount}
            >
              {({ input: { onChange, value } }) => {
                return (
                  <SliderRange
                    defaultValue={[2, 4]}
                    value={value || defaultValue}
                    min={1}
                    max={9}
                    step={1}
                    onChange={onChange}
                  />
                );
              }}
            </Field>
```

Note that the default value move from `initialValue` to component `value`.